packages/extension-ui/src/messaging.ts

```diff
@@ -30,7 +30,37 @@ interface Handler {
 
 type Handlers = Record<string, Handler>;
 
-const port = chrome.runtime.connect({ name: PORT_EXTENSION });
+async function wakeupBackground (): Promise<Error | null> {
+  try {
+    await chrome.runtime.sendMessage({ type: 'wakeup' });
+
+    return null;
+  } catch (cause) {
+    return cause instanceof Error ? cause : new Error(String(cause));
+  }
+}
+
+async function createPort (name: string, maxAttempts: number, delayMs: number): Promise<chrome.runtime.Port> {
+  let lastError: Error | null = null;
+
+  for (let attempt = 0; attempt < maxAttempts; attempt++) {
+    const error = await wakeupBackground();
+
+    if (error) {
+      lastError = error;
+      await new Promise((resolve) => setTimeout(resolve, delayMs));
+      continue;
+    }
+
+    const port = chrome.runtime.connect({ name });
+
+    return port;
+  }
+
+  throw new Error('Failed to create port after multiple attempts', { cause: lastError });
+}
+
+const port = await createPort(PORT_EXTENSION, 5, 1000);
```

Comment:
One question: Whats the significant of the delay value being 1000ms and the attempts being 5?

Reply:
Well the idea is that this wakes up the service_worker that's been terminated. If the "wake up" messsges are too many, there's something that's not working as it should, and if they are too far from eachother, it doesn't feel smooth, but those values are up to discussion

Reply:
Yea this makes sense!

Reply:
Happy to keep the values as is.